src/CUDAKernels.jl

```diff
@@ -13,6 +13,7 @@ CUDABackend(;prefer_blocks=false, always_inline=false) = CUDABackend(prefer_bloc
 
 export CUDABackend
 
+KernelAbstractions.unsafe_free!(x::CUDA.CuArray, stream::CUDA.CuStream=CUDA.stream()) = CUDA.unsafe_free!(x, stream)
```

Comment:
KA's `unsafe_free!` doesn't take a stream argument, so this looks wrong. Also, optimistically freeing on the _current_ stream might lead to invalid results.

Reply:
KA does not have the notion of streams at the moment and for example AMDGPU does not rely on streams for memory freeing.
We can remove the ability of passing `stream` argument.

> Also, optimistically freeing on the current stream might lead to invalid results.

The idea is that you can synchronize on the current task-local stream and safely free the array (which is what `KA.synchronize()` does).
Is it better to use the `default_stream`?

Reply:
> I can remove `stream` argument and instead always use `CUDA.stream`.

That seems better, otherwise people may write KA.jl code that works on CUDA.jl but isn't actually portable (because the `stream` argument).

cc @vchuravy